routers/repo/issue.go

```diff
@@ -734,6 +734,12 @@ func ViewIssue(ctx *context.Context) {
 		}
 		prConfig := prUnit.PullRequestsConfig()
 
+		if err := pull.CheckUserAllowedToMerge(ctx.User); err != nil {
```

Comment:
This could be rewritten as:
```go
ctx.Data["AllowMerge"] = ctx.Data["IsRepositoryWriter"]
if err := pull.CheckUserAllowedToMerge(ctx.User); err != nil {
    if !models.IsErrNotAllowedToMerge(err) {
        ctx.ServerError(err)
        return
    }
    ctx.Data["AllowMerge"] = false
}
```